### PR TITLE
Fix panic in deployment controller start

### PR DESCRIPTION
**Pull request by derekwaynecarr**
**What this PR does / why we need it**:
deployment controller can panic on startup

**Special notes for your reviewer**:
similar to https://github.com/kubernetes/kubernetes/pull/56941 but for deployments.

```release-note
fix potential panic on deployment controller start
```

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
fyi @ironcladlou @ncdc @gnufied 

have not had enough time to test this, but seeing similar symptoms as referenced pr but for deployments.

File: pkg/controller/deployment/deployment_controller.go [hunk 1]
```diff
@@ -133,15 +133,15 @@ func NewDeploymentController(dInformer extensionsinformers.DeploymentInformer, r
...
 
-	dc.syncHandler = dc.syncDeployment
-	dc.enqueueDeployment = dc.enqueue
-
 	dc.dLister = dInformer.Lister()
 	dc.rsLister = rsInformer.Lister()
 	dc.podLister = podInformer.Lister()
 	dc.dListerSynced = dInformer.Informer().HasSynced
 	dc.rsListerSynced = rsInformer.Informer().HasSynced
 	dc.podListerSynced = podInformer.Informer().HasSynced
```
Comment (ironcladlou) [inline pkg/controller/deployment/deployment_controller.go]: Looks like this is one of the bits of work that got lost when https://github.com/kubernetes/kubernetes/pull/51492 was closed. As a defensive maneuver, inline everything possible into the DeploymentController instance creation.

**Review by ironcladlou — CHANGES_REQUESTED** [review]

**Comment by ironcladlou** [comment]
Another instance of https://github.com/kubernetes/kubernetes/issues/51013

File: pkg/controller/deployment/deployment_controller.go (hunk 1 above)
Comment (ironcladlou) [reply pkg/controller/deployment/deployment_controller.go]: And anything that's not inlined, move to lines preceding calls to `AddEventHandler`.

**Comment by derekwaynecarr** [comment]
@ironcladlou -- think i captured your feedback, PTAL.

**Comment by derekwaynecarr** [comment]
closing per discussion with @deads2k to fix this another way.

**Pull request closed without merging**